force-app/main/default/classes/IBMLanguageTranslatorV3.cls

```diff
@@ -243,35 +275,142 @@ public class IBMLanguageTranslatorV3 extends IBMWatsonService {
     builder.addHeader('Accept', 'application/json');
-    builder.addHeader('X-IBMCloud-SDK-Analytics', 'service_name=language_translator;service_version=V3;operation_id=listModels');
-    Map<String, String> requestHeaders = (listModelsOptions != null) ? listModelsOptions.requestHeaders() : null;
+    builder.addHeader('X-IBMCloud-SDK-Analytics', 'service_name=language_translator;service_version=V3;operation_id=listDocuments');
+    Map<String, String> requestHeaders = (listDocumentsOptions != null) ? listDocumentsOptions.requestHeaders() : null;
     if (requestHeaders != null && requestHeaders.size() > 0) {
       for (String name : requestHeaders.keySet()) {
         builder.addHeader(name, requestHeaders.get(name));
       }
     }
     builder.query('version', versionDate);
-    if (listModelsOptions != null && listModelsOptions.source() != null) {
-      builder.query('source', listModelsOptions.source());
+
+    return (IBMLanguageTranslatorV3Models.DocumentList) createServiceCall(builder.build(), IBMLanguageTranslatorV3Models.DocumentList.class);
+  }
+
+  /**
+   * Translate document.
+   *
+   * Submit a document for translation. You can submit the document contents in the `file` parameter, or you can
+   * reference a previously submitted document by document ID.
+   *
+   * @param translateDocumentOptions the {@link IBMLanguageTranslatorV3Models.TranslateDocumentOptions} containing the options for the call
+   * @return the {@link IBMLanguageTranslatorV3Models.DocumentStatus} with the response
+   */
+  public IBMLanguageTranslatorV3Models.DocumentStatus translateDocument(IBMLanguageTranslatorV3Models.TranslateDocumentOptions translateDocumentOptions) {
+    IBMWatsonValidator.notNull(translateDocumentOptions, 'translateDocumentOptions cannot be null');
+    IBMWatsonRequestBuilder builder = IBMWatsonRequestBuilder.httpPost(getEndPoint() + '/v3/documents');
+    builder.addHeader('Accept', 'application/json');
+    builder.addHeader('X-IBMCloud-SDK-Analytics', 'service_name=language_translator;service_version=V3;operation_id=translateDocument');
+    Map<String, String> requestHeaders = (translateDocumentOptions != null) ? translateDocumentOptions.requestHeaders() : null;
+    if (requestHeaders != null && requestHeaders.size() > 0) {
+      for (String name : requestHeaders.keySet()) {
+        builder.addHeader(name, requestHeaders.get(name));
+      }
     }
-    if (listModelsOptions != null && listModelsOptions.target() != null) {
-      builder.query('target', listModelsOptions.target());
+    builder.query('version', versionDate);
+    IBMWatsonMultipartBody.Builder multipartBuilder = new IBMWatsonMultipartBody.Builder();
+    multipartBuilder.setType(IBMWatsonMultipartBody.FORM);
+    IBMWatsonRequestBody fileBody = IBMWatsonRequestBody.create(translateDocumentOptions.file(), translateDocumentOptions.fileContentType());
+    multipartBuilder.addFormDataPart('file', null, fileBody);
```

Comment:
We want to send a filename for the file since the filename shows up when we get the document status. I'm not sure if this is set somewhere else.